2.x port

zachmullen opened the pull request.

msmolens commented:
Nice!

I tried to test by running `vagrant up` after setting:

    girder_version: master
    covalic_version: 2.x-port

but get an error:

    TASK [girder : set the mongodb server ip in girder local config] ***************
    fatal: [default]: FAILED! => {"changed": false, "failed": true, "msg": "Destination /opt/covalic/girder/girder/conf/girder.local.cfg does not exist !", "rc": 257}

Additionally, the Girder branch in `.travis.yml` should be updated.

zachmullen commented:
The vagrant install is fixed for me, and I've switched to using the modern version of webroot route management via the system setting. Also upgraded to the latest girder ansible module and cleaned up some of our girder ansible role.

msmolens commented on web_external/views/body/SubmissionView.js:
```diff
@@ -60,32 +79,33 @@ covalic.views.SubmissionView = covalic.View.extend({
                         _id: this.submission.get('jobId')
                     }).on('g:fetched', function () {
-                        if (this.job.get('status') === girder.jobs_JobStatus.ERROR) {
+                        if (this.job.get('status') === JobStatus.ERROR) {
```
The `fetch()` should not be chained anymore: 
Uncaught TypeError: this.job.get is not a function`
Perhaps there are more problems like this one?

msmolens commented on plugin.cmake:
```diff
@@ -8,7 +8,10 @@ add_python_style_test(python_static_analysis_covalic
 )
+
+add_puglint_test(
+    covalic "${PROJECT_SOURCE_DIR}/plugins/${PLUGIN}/web_external/templates")
```
`PLUGIN` is not defined in this file, but it could reduce duplication if we define it!

msmolens commented:
Following https://github.com/girder/girder/pull/1731, `vagrant up` fails at step `girder : Build and install web client code` with a huge error output, the relevant piece being `SyntaxError: Unexpected token ...`.

zachmullen commented:
Lame, thought that worked on latest node v4, guess it's too new of a language feature. I'll change it in girder to use `concat` instead.

zachmullen commented:
https://github.com/girder/girder/pull/1735

msmolens commented:
Great, provisioning succeeds. But now browsing to the root gives a 404 error. The admin interface on /girder works.

msmolens commented:
Thanks for checking on your machine. I now see Covalic in the root.

msmolens commented on web_external/views/body/SubmissionView.js:
```diff
@@ -1,23 +1,42 @@
-covalic.views.SubmissionView = covalic.View.extend({
+import _ from 'underscore';
+import { getCurrentUser } from 'girder/auth';
+import { AccessType, DATE_SECOND } from 'girder/constants';
```
`DATE_SECOND` is in `girder/misc`. Is there any way to detect errors like this other than at runtime?

zachmullen replied:
Unfortunately runtime is the only way for this style of imports. It'll detect bad paths statically, but not bad symbols.

msmolens commented:
I tested submitting to a phase and the jobs never move beyond "Queued." Can you try this?

zachmullen commented:
Full end-to-end is working for me now.

msmolens commented:
Nice! But I still can't get throw the basic workflow:
- As an admin, clicking "View all submissions" on the phase page gives an error:

> Uncaught TypeError: e.get(...).split is not a function
    at _getShortLog (covalic.min.js:2)

- A trivial submission workflow as ad admin fails with an error:
  - The "log output" is empty
  - The job timeline widget isn't rendered
  - The console shows an error:
> Uncaught TypeError: this.text(...).toLowerCase is not a function

- Unimportant, but on my Windows system if I change the Vagrantfile to use ansible_local, `pip: extra_args="-e" name="/opt/girder_worker[girder_io,docker]"` fails, even though I've been able to test like this up to this point:

> {"changed": false, "cmd": "/usr/local/bin/pip2 install -e /opt/girder_worker[girder_io docker]", "failed": true, "msg": "\n:stderr: Invalid requirement: 'docker]'\nTraceback (most recent call last):\n  File \"/usr/local/lib/python2.7/dist-packages/pip/req/req_install.py\", line 82, in __init__\n    req = Requirement(req)\n  File \"/usr/local/lib/python2.7/dist-packages/pip/_vendor/packaging/requirements.py\", line 96, in __init__\n    requirement_string[e.loc:e.loc + 8]))\nInvalidRequirement: Invalid requirement, parse error at \"']'\"\n\n"}

Separating the list into two steps makes it work:
>pip: extra_args="-e" name="/opt/girder_worker[girder_io]"
...
pip: extra_args="-e" name="/opt/girder_worker[docker]"

zachmullen commented:
Ok, all those errors should now be fixed.

msmolens commented:
Cool, it works for me now.
Also relevant, to me at least, but unrelated to this branch, is that the arguments passed to the scoring container are different from Covalic circa the ISBI Challenge 2016. This means that the scoring container used for the ISBI challenge is no longer compatible. The relevant change to the default scoring container is https://github.com/InsightSoftwareConsortium/covalic/pull/24. cc @brianhelba
